### DISCUSSION TITLE
How to get last parent's info in multi hop connection

### DISCUSSION THREAD

**Discussion by ksathara** [openthread/openthread#10143] [Q&A] [answered]
Hi,

I have multiple sensors (SEDs) and one or more Hosts (FTDs) in our application. Based on the distance or other similar issues, there is a possibility that a sensor may connect to it's paired host via another router/leader. I want to find out the details of the paired hosts RLOC16, ext address etc. I tried parent info from CLI, but that reports the immediate connection, not the last hop. Is there a possibility to get this information?

Cheers,

Kaushalya

**Comment by jwhui** [comment]
By design, Thread network state is distributed across the network - no single device maintains topology information for all Thread devices in the network.

It is possible to query topology information from other nodes. For example, you can use the [`meshdiag childtable`](https://github.com/openthread/openthread/tree/main/src/cli#meshdiag-childtable-router-rloc16) command to query the child table from a given router.

**Reply by ksathara** [reply]
Thanks Jonathan, When I tried meshdiag childtable with my SED or FTD it responds invalid command. I use SDK 2.3.0, may be this command is not implemented?
Cheers,
Kaushalya

**Reply by jwhui** [reply, marked as answer]
Most likely your OpenThread snapshot does not include:
- https://github.com/openthread/openthread/pull/8866

**Reply by abtink** [reply]
Another command to try out [`meshdiag topology children`](https://github.com/openthread/openthread/tree/main/src/cli#meshdiag-topology-ip6-addrschildren) (if supported on device/build). it uses older netdiag TLV definitions, so may be supported on more builds (also works in case the parent is running older versions of Thread stack).

**Reply by ksathara** [reply]
Hi Abtink,

Many thanks for your contribution. I have already tried this, but unfortunately this also not supported in my current build. When I check all commands supported, I cant see any meshdiag commands. May be I have to upgrade my SDK as my current SDK is quite old. I am using Nordic Connect SDK 2.3.0.

When I tried 'version' command on CLI, I get 'OPENTHREAD/2060039bb-dirty;' I dont know how old this versions is. May be you can tell me?

Cheers,
Kaushalya

**Reply by jwhui** [reply]
https://github.com/openthread/openthread/commit/2060039bb
